public/js/NeighbourPlot.js

```diff
@@ -27,23 +27,25 @@ d3.selection.prototype.moveToFront = function() {
  * Clicking on scatterplot points will trigger updates of the neighbour
  * image gallery and corresponding sample lineplot.
  *
+ * The width and height of the plot is calculated based on the size of the
+ * plot's containing DIV. The width is taken from the DIV, and the height
+ * calculated from that width such that the plot will have a 16:9 aspect ratio.
+ *
  * @constructor
  * @param {array} sampleData - The sample data for the screen. Each element
  *     in the array is an instance of a Sample document.
  * @param {string} element - The ID of the target div for this plot.
- * @param {LinePlot} lineplot - LinePlot that will update when scatterplot
- *     points are clicked.
- * @param {NeighbourImages} neighbourImages - NeighbourImages object that will
  * update when scatterplot points are clicked.
  */
-function NeighbourPlot(sampleData, element, lineplot, neighbourImages) {
-    this.sampleData = sampleData;
-    this.lineplot = lineplot;
-    this.neighbourImages = neighbourImages;
+function NeighbourPlot(sampleData, element) {
+    var self = this;
 
+    this.sampleData = sampleData;
     this.element = element;
-    this.fullWidth = 700;
-    this.fullHeight = 500;
+
+    this.fullWidth = $(this.element).width();
+    this.fullHeight = Math.round(this.fullWidth * (9/16));
```

Comment:
Reminds me of something I saw a while back: "there's two kinds of people in this world: those who understand resolutions and aspect ratios; and morons." =D I still marvel at all the *brand new planes* with 4:3 movies shown stretched on a 16:9 screen. Who the fuck is responsible for these things???

Ok that was slightly off topic. =P